ethcore/src/client/client.rs

```diff
@@ -187,6 +186,20 @@ impl Client {
 			try!(db.write(batch).map_err(ClientError::Database));
 		}
 
+		trace!("Cleanup journal: DB Earliest = {:?}, Latest = {:?}", state_db.journal_db().earliest_era(), state_db.journal_db().latest_era());
+		if let (Some(earliest), Some(latest)) = (state_db.journal_db().earliest_era(), state_db.journal_db().latest_era()) {
+			if latest > earliest && latest - earliest > config.history {
+				let mut era = earliest;
+				while era <= latest - config.history {
```

Comment:
`for era in earliest..(latest - config.history + 1) {` is a bit less error-prone.